pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -652,9 +652,29 @@ def process_tokens(self, tokens: list[tokenize.TokenInfo]) -> None:
         trailing_comma_tuple_enabled_for_file = self.linter.is_message_enabled(
             "trailing-comma-tuple"
         )
+        trailing_comma_tuple_enabled_once: bool = trailing_comma_tuple_enabled_for_file
         # Process tokens and look for 'if' or 'elif'
         for index, token in enumerate(tokens):
             token_string = token[1]
+            if (
```

Comment:
What do you think about factoring this out into a helper method? It's almost 20 lines with comments.

Reply:
I do think we need to make this kind of optimization where we do not compute things when a message is disabled easier to use because the faster computation is the one you never do.

If we create an helper then we'd probably want to parametrize "trailing-comma-tuple" / "R1707" so it can be useful elsewhere with other messages (then we should be using the message store to get old name of the message if a message was ever renamed ?).

We have the "perfect function" for this already with ``self.linter.is_message_enabled(`` applied on a line (barring some optimization to do here related to not launching it on each token as they are multiple token per line). It felt very heavy handed in this case so I did something very specific without check for old name / and without exact check of enabling or not. 

To be honest I made a lot of assumptions without ever doing a benchmark. One way to keep this clean and dry would be to make the message store better for this use case (token checks, and probably enabled but not absolutely sure it still is?).

Reply:
No worries, we can merge as is. I wasn't suggesting reusing it elsewhere, I just meant hoisting it out of this current location into a few lines up in the same file.